repos/fdbt-site/src/pages/api/manageTimeRestriction.ts

```diff
@@ -0,0 +1,205 @@
+                timeBands: [
+                    {
+                        startTime: removeAllWhiteSpace(startTimeInputs),
+                        endTime: removeAllWhiteSpace(endTimeInputs),
+                    },
+                ],
+            });
+        }
+    });
+    return fullTimeRestrictions;
+};
+
+export const collectErrors = (refinedName: string, fullTimeRestrictions: FullTimeRestriction[]): ErrorInfo[] => {
+    const errors: ErrorInfo[] = [];
+
+    if (!refinedName) {
+        errors.push({
+            errorMessage: `Time restriction name is required.`,
+            id: 'time-restriction-name',
+            userInput: refinedName,
+        });
+    }
+
+    if (!fullTimeRestrictions.length) {
+        errors.push({ errorMessage: `You must select at least one day.`, id: 'time-restriction-days' });
+    }
+
+    fullTimeRestrictions.forEach((fullTimeRestriction) => {
+        fullTimeRestriction.timeBands.forEach((timeBand, index) => {
+            if (timeBand.startTime && !isValid24hrTimeFormat(timeBand.startTime)) {
+                if (timeBand.startTime === '2400') {
+                    errors.push({
+                        errorMessage: '2400 is not a valid fullTimeRestriction. Use 0000.',
+                        id: `start-time-${fullTimeRestriction.day}-${index}`,
+                        userInput: timeBand.startTime,
+                    });
+                } else {
+                    errors.push({
+                        errorMessage: 'Time must be in 24hr format.',
+                        id: `start-time-${fullTimeRestriction.day}-${index}`,
+                        userInput: timeBand.startTime,
+                    });
+                }
+            }
+
+            if (timeBand.endTime && !isValid24hrTimeFormat(timeBand.endTime)) {
+                if (timeBand.endTime === '2400') {
+                    errors.push({
+                        errorMessage: '2400 is not a valid input. Use 0000.',
+                        id: `end-time-${fullTimeRestriction.day}-${index}`,
+                        userInput: timeBand.endTime,
+                    });
+                } else {
+                    errors.push({
+                        errorMessage: 'Time must be in 24hr format.',
+                        id: `end-time-${fullTimeRestriction.day}-${index}`,
+                        userInput: timeBand.endTime,
+                    });
+                }
+            }
+
+            if (
+                isValid24hrTimeFormat(timeBand.startTime) &&
+                isValid24hrTimeFormat(timeBand.endTime) &&
+                timeBand.startTime === timeBand.endTime
+            ) {
+                errors.push(
+                    {
+                        errorMessage: 'Start time and end time cannot be the same.',
+                        id: `start-time-${fullTimeRestriction.day}-${index}`,
+                        userInput: timeBand.startTime,
+                    },
+                    {
```

Comment:
Why is there 2 errors here?

Reply:
Ah yep agreed we could just have one, have changed that.